packages/ui/src/lib/button/Button.svelte

```diff
@@ -1,3 +1,49 @@
-<button class="btn bg-core-blue-600 p-4">
-    <slot></slot>
-</button>
+<script lang="ts">
+	import classNames from 'classnames';
+
+	export let style: 'primary' | 'secondary' | 'ghost' = 'primary';
+	export let size: 'xs' | 'sm' | 'base' | 'lg' | 'xl' = 'base';
+	export let href: string | undefined = undefined;
+	export let type: 'button' | 'submit' = 'button';
+
+	const styleClasses = {
+		primary: 'bg-core-grey-500 border-b-4 border-core-blue-600 text-white hover:bg-core-grey-400',
+		secondary: 'bg-white border-b-4 border-core-blue-600 hover:bg-core-grey-100',
+		ghost: 'bg-transparent hover:bg-core-grey-100 dark:text-white dark:hover:bg-core-grey-400'
+	};
+
+	const sizeClasses = {
+		xs: 'text-xs',
+		sm: 'text-sm',
+		base: 'text-base',
+		lg: 'text-lg',
+		xl: 'text-xl'
+	};
+
+	let buttonClass: string;
+
+	$: buttonClass = classNames(
+		'px-4 py-2 inline-flex justify-center',
+		sizeClasses[size],
+		styleClasses[style],
+		$$props.disabled && 'cursor-not-allowed text-core-grey-300 border-core-grey-200 opacity-50',
```

Comment:
I recommend making the _disabled_ prop explicit `export let disabled = false` so it's clear that it's part of the component interface. I think it's needed for TS and some other tools to validate usage.

Reply:
It also won't work for `<a>` tag. You'd need to add CSS _`pointer-events: none;`_ to disable it (or equivalent Tailwind). Not an issue if we remove the anchors from the component.